src/main/java/org/gooru/insights/services/BaseESServiceImpl.java

```diff
@@ -296,23 +299,28 @@ private void groupConcat(List<Map<String,Object>> dataList,List<Map<String,Objec
 			buildBuckets(traceId,indices,requestParamsDTO, searchRequestBuilder,metricsName,validatedData,filterData, userFilter);
 			hasAggregate = true;
 		
+		}	else if (validatedData.get(Hasdatas.HAS_AGGREGATE.check())) {
+			searchRequestBuilder.setNoFields();
+			buildAggregation(traceId,indices,requestParamsDTO, searchRequestBuilder,metricsName,validatedData,filterData, userFilter);
+			hasAggregate = true;
+		
 		}  else {
 			Set<String> filterFields = new HashSet<String>();
-			if(validatedData.get(Hasdatas.HAS_FEILDS.check())){
-				if(!requestParamsDTO.getFields().equalsIgnoreCase(APIConstants.WILD_CARD)){
-			dataKey = EsSources.FIELDS.esSource();
-			String fields = getBusinessLogicService().esFields(indices,requestParamsDTO.getFields());
-			filterFields = getBaseAPIService().convertStringtoSet(fields);
-			}else{
-					for(String field : getBaseConnectionService().getDefaultFields().get(indices).split(APIConstants.COMMA)){
-						searchRequestBuilder.addField(field);	
+			if (validatedData.get(Hasdatas.HAS_FEILDS.check())) {
+				if (!requestParamsDTO.getFields().equalsIgnoreCase(APIConstants.WILD_CARD)) {
+					dataKey = EsSources.FIELDS.esSource();
+					String fields = getBusinessLogicService().esFields(indices, requestParamsDTO.getFields());
+					filterFields = getBaseAPIService().convertStringtoSet(fields);
+				} else {
```

Comment:
for (String field : filterFields) {
+					searchRequestBuilder.addField(field);
+				}

I think above iteration method should be inside  the if condition. // if(!requestParamsDTO.getFields().equalsIgnoreCase(APIConstants.WILD_CARD)){